### ISSUE TITLE
Getting "No browser found on device, abort login" on startup

### ISSUE THREAD

**Issue by CharlesMcD** [microsoft/appcenter-sdk-android#1481]
### **Description**

I get this error when the sdk tries to open a browser to authenticate. (I'm using UpdateTrack.PRIVATE)

```
I/AppCenter: App Center SDK configured successfully.
.......
D/AppCenterDistribute: Resume distribute workflow...
D/AppCenterDistribute: InstallerPackageName=null
D/AppCenterDistribute: No token, need to open browser to url=https://install.appcenter.ms/apps/4c671dd5-0be2-4c24-ae09-42d252e4d0e6/private-update-setup/?release_hash=17be617c2303b565829055a0768c75df801e07b74e244979378903a473d0eec4&redirect_id=***OBFUSCATED**&&redirect_scheme=appcenter&request_id=9709a2b5-e6b0-457b-8854-9186ec7b7f0f&platform=Android&enable_failure_redirect=true&install_id=7439b8f6-b1cb-4c76-b274-80b55b4b9b97
V/FA: Activity resumed, time: 569295518
E/AppCenterDistribute: No browser found on device, abort login.
```

But Google chrome is the default browser

![device-2020-12-02-163314](https://user-images.githubusercontent.com/6673075/100934236-7b2d6280-34bc-11eb-8210-40f5a9c86e03.png)

### **Repro Steps**

By following the instructions here : https://docs.microsoft.com/en-us/appcenter/sdk/distribute/android
Here's what's in my build.gradle file : 

```
implementation "com.microsoft.appcenter:appcenter-distribute:4.0.0"
prodImplementation "com.microsoft.appcenter:appcenter-distribute-play:4.0.0"
```

And in `class MyApplication : Application()`
```
override fun onCreate() {
    super.onCreate()

    AppCenter.setLogLevel(Log.VERBOSE)
    Distribute.setEnabledForDebuggableBuild(true)
    Distribute.setUpdateTrack(UpdateTrack.PRIVATE)
    AppCenter.start(this, appCenterAppSecret, Distribute::class.java)
}
```

`appCenterAppSecret` is correctly set.

### **Details**

1. Which SDK version are you using?
    - 4.0.0 and I experience the same thing on 3.3.1
2. Which OS version did you experience the issue on?
    - Android 11
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - Pixel 4 and Pixel 4a physical device
4. What third party libraries are you using?
    - Nothing
5. Please enable verbose logging for your app using `AppCenter.setLogLevel(Log.VERBOSE)` before your call to `AppCenter.start(...)` and include the logs here:
```
2020-12-02 16:11:43.430 16371-16371/  I/AppCenter: App Center SDK configured successfully.
2020-12-02 16:11:43.430 16371-16371/  D/AppCenter: Cannot read instrumentation variables in a non-test environment.
2020-12-02 16:11:43.431 16371-16371/  I/System.out: com.microsoft.appcenter.utils.async.DefaultAppCenterFuture@46bcb06
2020-12-02 16:11:43.435 16371-16464/  D/AppCenter: Loaded stored sessions: {1606937129072=1606937129072//1606937129071, 1606937407829=1606937407829//1606937407824, 1606939024542=1606939024542//1606939024541, 1606940726286=1606940726286//1606940726284, 1606940901741=1606940901741//1606940901739, 1606941238836=1606941238836//1606941238833, 1606942191206=1606942191206//1606942191204, 1606942220165=1606942220165//1606942220163, 1606942257158=1606942257158//1606942257156, 1606942676704=1606942676704//1606942676701}
2020-12-02 16:11:43.442 16371-16468/  D/AppCenter: Network 105 is available.
2020-12-02 16:11:43.442 16371-16468/  D/AppCenter: Network has been connected.
2020-12-02 16:11:43.446 16371-16464/  I/AppCenter: Changed maximum database size to 10485760 bytes.
2020-12-02 16:11:43.446 16371-16464/  D/AppCenter: addGroup(group_core)
2020-12-02 16:11:43.447 16371-16464/  D/AppCenter: checkPendingLogs(group_core) pendingLogCount=0 batchTimeInterval=3000
2020-12-02 16:11:43.448 16371-16464/  D/AppCenter: App Center initialized.
2020-12-02 16:11:43.449 16371-16464/  D/AppCenter: removeGroup(group_distribute)
2020-12-02 16:11:43.449 16371-16464/  D/AppCenter: removeGroup(group_distribute/one)
2020-12-02 16:11:43.449 16371-16464/  D/AppCenter: addGroup(group_distribute)
2020-12-02 16:11:43.450 16371-16464/  D/AppCenter: checkPendingLogs(group_distribute) pendingLogCount=0 batchTimeInterval=3000
2020-12-02 16:11:43.450 16371-16464/  D/AppCenter: addGroup(group_distribute/one)
2020-12-02 16:11:43.451 16371-16464/  D/AppCenter: checkPendingLogs(group_distribute/one) pendingLogCount=0 batchTimeInterval=3000
2020-12-02 16:11:43.451 16371-16464/  D/AppCenterDistribute: Distribute workflow will be resumed on activity resume event.
2020-12-02 16:11:43.451 16371-16464/  I/AppCenter: Distribute service started from application.
2020-12-02 16:11:43.455 16371-16464/  D/AppCenter: Storing a log to the Persistence database for log type startService with flags=1
2020-12-02 16:11:43.464 16371-16464/  D/AppCenter: Stored a log to the Persistence database for log type startService with databaseId=11
2020-12-02 16:11:43.464 16371-16464/  D/AppCenter: enqueue(group_core) pendingLogCount=1
2020-12-02 16:11:43.464 16371-16464/  D/AppCenter: checkPendingLogs(group_core) pendingLogCount=1 batchTimeInterval=3000
2020-12-02 16:11:43.464 16371-16464/  I/AppCenterDistribute: Distribute service has already been enabled.
2020-12-02 16:11:43.510 16371-16448/  V/FA: onActivityCreated
2020-12-02 16:11:43.652 16371-16371/  D/AppCenterDistribute: Resetting workflow on entering foreground.
2020-12-02 16:11:44.269 16371-16371/  D/AppCenterDistribute: Resume distribute workflow...
2020-12-02 16:11:44.269 16371-16371/  D/AppCenterDistribute: InstallerPackageName=null
2020-12-02 16:11:44.270 16371-16371/  D/AppCenterDistribute: No token, need to open browser to url=https://install.appcenter.ms/apps/4c671dd5-0be2-4c24-ae09-42d252e4d0e6/private-update-setup/?release_hash=17be617c2303b565829055a0768c75df801e07b74e244979378903a473d0eec4&redirect_id=***OBFUSCATED***&redirect_scheme=appcenter&request_id=9709a2b5-e6b0-457b-8854-9186ec7b7f0f&platform=Android&enable_failure_redirect=true&install_id=7439b8f6-b1cb-4c76-b274-80b55b4b9b97
2020-12-02 16:11:44.273 16371-16371/  E/AppCenterDistribute: No browser found on device, abort login.
```

**Comment by olmisa** [comment]
Hello @CharlesMcD ,
Thank you for contacting us. As I understand this issue should be fixed [here](https://github.com/microsoft/appcenter-sdk-android/pull/1475).
I hope the next release should solve your problem.
Thanks

**Comment by AnastasiaKubova** [comment]
The fix has been released in version `4.1.0`. Please update the App Center SDK to the [latest version](https://github.com/microsoft/appcenter-sdk-android/releases/tag/4.1.0) and let me know if this issue still persists.
